### PR TITLE
Nix devcontainer

### LINKED ISSUE #784: Build a `devcontainer` for this project to ease getting a local env stood up
It would be nice if there was a [devcontainer](https://code.visualstudio.com/docs/devcontainers/containers) (nix/arch)? people could use with all the tooling installed into it to run thru the setup.

### PR DESCRIPTION
**Pull request by whazor**
Resolves #784

### CODE REVIEW

**Review by onedr0p — COMMENTED** [review]
Would it be possible to nest some/all of the config files under a `.devcontainer/` dir like done in https://github.com/renovatebot/renovate/tree/main/.devcontainer ?

**Comment by whazor** [comment]
It is a bit slow, but it works: https://github.com/whazor/flux-cluster-template

It does download one gigabyte of dependencies, so I guess it is fast in that sense

**Comment by onedr0p** [comment]
Awesome, I am giving it a shot now

File: .devcontainer/devenv.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ pkgs, ... }:
+
+{
+  packages = with pkgs; [
```
Comment (onedr0p) [inline .devcontainer/devenv.nix:4]: Might want to include dig, ping, grep and some other generic tools?

**Comment by onedr0p** [comment]
What am I doing wrong here 😄 I opened vscode and went the shell, however there's no cli tools :/

```
vscode ➜ /workspaces/flux-cluster-template (devcontainer) $ ping
bash: ping: command not found
vscode ➜ /workspaces/flux-cluster-template (devcontainer) $ ansible
bash: ansible: command not found
vscode ➜ /workspaces/flux-cluster-template (devcontainer) $ age
bash: age: command not found
vscode ➜ /workspaces/flux-cluster-template (devcontainer) $ 
```

**Comment by whazor** [comment]
Ahhh, 505 is my Mac user ID.

**Comment by whazor** [comment]
It does not work for me in Colima, but it does work for me with Rancher Desktop. So I think it depends on the volume mounting method that you are using. SSHFS or the internal Mac one, which messes up the user IDs.

**Comment by onedr0p** [comment]
With the recent changes it's much easier to get started, there's a lot less CLI tools needed. Closing this. Thanks for the PR nonetheless

**Pull request closed without merging**